apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -366,12 +369,41 @@ export default class CoreLibrary {
     return speechBubbles[speechBubbles.length - 1];
   }
 
+  reachedSpriteMax() {
+    const numSprites = this.getNumberOfSprites();
+    return numSprites >= MAX_NUM_SPRITES;
+  }
+
+  // When the total number of sprites is equal to SPRITE_LIMIT_WARNING,
+  // a workspace alert warning is displayed
+  // Note that SPRITE_LIMIT_WARNING = MAX_NUM_SPRITES - 1
+  warnIfAtSpriteLimit() {
+    const numSprites = this.getNumberOfSprites();
+    if (numSprites === SPRITE_LIMIT_WARNING) {
+      getStore().dispatch(
+        displayWorkspaceAlert(
+          workspaceAlertTypes.warning,
+          /* display warning when user exceeds SPRITE_LIMIT_WARNING */
+          msg.spriteLimitReached({limit: SPRITE_LIMIT_WARNING}),
+          /* bottom */ true
+        )
+      );
+    }
+  }
+
   /**
    * Adds the specified sprite to the native sprite map
    * @param {Sprite} sprite
    * @returns {Number} A unique id to reference the sprite.
    */
   addSprite(opts) {
+    // this function returns early if total number of sprites equals MAX_NUM_SPRITES
+    if (this.reachedSpriteMax()) {
+      return;
+    }
+    // this function dispatches a workspace alert if the total number of sprites
+    // equals SPRITE_LIMIT_WARNING
+    this.warnIfAtSpriteLimit();
```

Comment:
Minor nit, but since this set of code is repeated a few times, I don't think it needs to be commented this heavily in each case - more potential for one comment to get out of sync with others which could lead to confusion down the road. As long as the original functions are well commented (which `warnIfAtSpriteLimit` seems to be; may want to add a quick comment to `reachedSpriteMax`, though it is pretty self-explanatory 🙂) I don't think you need to include too many comments here.

Reply:
Thanks - makes sense to remove repetitive comments. Made the updates!